components/automate-cli/cmd/chef-automate/cmdUtils.go

```diff
@@ -0,0 +1,428 @@
+		const remoteService string = CONST_POSTGRESQL
+		nodeIps, err := preCmdExecCheck(c.NodeMap.Postgresql, c.SshUtil, c.NodeMap.Infra, remoteService, timestamp, writer)
+		if err != nil {
+			return err
+		}
+
+		err = c.CmdExecOnNodes(c.NodeMap.Postgresql.CmdInputs, nodeIps, remoteService, timestamp, writer)
+	case c.NodeMap.Opensearch.CmdInputs.NodeType:
+		const remoteService string = CONST_OPENSEARCH
+		nodeIps, err := preCmdExecCheck(c.NodeMap.Opensearch, c.SshUtil, c.NodeMap.Infra, remoteService, timestamp, writer)
+		if err != nil {
+			return err
+		}
+
+		err = c.CmdExecOnNodes(c.NodeMap.Opensearch.CmdInputs, nodeIps, remoteService, timestamp, writer)
+	default:
+		return errors.New("Missing or Unsupported flag")
+	}
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (c *CmdUtilImpl) CmdExecOnNodes(input *CmdInputs, nodeIps []string, remoteService string, timestamp string, cliWriter *cli.Writer) error {
+	resultChan := make(chan CmdResult, len(nodeIps))
+	originalSSHConfig := c.SshUtil.getSSHConfig()
+	inputFiles := input.InputFiles
+	outputFiles := input.Outputfiles
+	command := input.Cmd
+
+	inputFileToOutputFileMap := map[string]string{}
+	for _, file := range inputFiles {
+		configFile := remoteService + "_" + timestamp + "_" + file
+		inputFileToOutputFileMap[file] = configFile
+	}
+
+	for _, hostIP := range nodeIps {
+		newSSHConfig := &SSHConfig{
+			sshUser:    originalSSHConfig.sshUser,
+			sshPort:    originalSSHConfig.sshPort,
+			sshKeyFile: originalSSHConfig.sshKeyFile,
+			hostIP:     hostIP,
+		}
+		newSSHUtil := NewSSHUtil(newSSHConfig)
+
+		printConnectionMessage(remoteService, hostIP, cliWriter)
+
+		go c.RemoteJobs(command, inputFileToOutputFileMap, outputFiles, remoteService, input.ErrorCheck, newSSHUtil, resultChan)
+	}
+
+	for i := 0; i < len(nodeIps); i++ {
+		result := <-resultChan
+
+		if i == 0 {
+			writer.StopSpinner()
+			cliWriter.Println("=====================================================")
+		}
+
+		printOutput(remoteService, result, outputFiles, cliWriter)
+
+		if i < len(nodeIps)-1 {
+			cliWriter.Println("=====================================================")
+			writer.StartSpinner()
+		}
+	}
+
+	close(resultChan)
+	return nil
+}
+
+// RemoteJobs function will run all remote jobs.
+func (c *CmdUtilImpl) RemoteJobs(command string, inputFiles map[string]string, outputFiles []string, remoteService string, errorCheck bool, newSSHUtil SSHUtil, resultChan chan CmdResult) {
+	rc := CmdResult{newSSHUtil.getSSHConfig().hostIP, []string{}, "", nil}
+	if len(inputFiles) != 0 {
+		for inputFile, outputFile := range inputFiles {
+			err := newSSHUtil.copyFileToRemote(inputFile, outputFile, false)
+			if err != nil {
+				rc.Error = err
+				resultChan <- rc
+				return
+			}
+		}
+	}
+
+	output, err := newSSHUtil.connectAndExecuteCommandOnRemote(command, true)
+	if err != nil {
+		rc.Error = err
+		resultChan <- rc
+		return
+	}
+
+	if len(outputFiles) != 0 {
```

Comment:
We are doing multiple things in one function (copy file to remote, execute the command on remote and copy file from remote). I think it would be better to break this function into multiple smaller functions as required.

Reply:
Discussed